src/tests/example.spec.js

```diff
@@ -1,27 +1,193 @@
+  });
 
-        await expect(inventoryPage.headerTitle).toBeVisible();
+  test("Perform login", async ({ inventoryPage }) => {
+    await expect(inventoryPage.headerTitle).toBeVisible();
 
-        expect(await inventoryPage.inventoryItems.count()).toBeGreaterThanOrEqual(1);
-    });
+    expect(await inventoryPage.inventoryItems.count()).toBeGreaterThanOrEqual(
+      1
+    );
+  });
+
+  
+  test("Add and remove product from the cart", async ({
+    inventoryPage,
+    shopingCartPage,
+  }) => {
+    await inventoryPage.addItemToCartById(0);
+    expect(await inventoryPage.getNumberOfItemsInCart()).toBe("1");
+
+    await inventoryPage.shopingCart.click();
+    expect(await shopingCartPage.cartItems.count()).toBeGreaterThanOrEqual(1);
+
+    await shopingCartPage.removeCartItemById(0);
+    await expect(shopingCartPage.cartItems).not.toBeAttached();
+  });
 
-    test('Add and remove product from the cart', async ({ loginPage, inventoryPage, shopingCartPage }) => {
-        await loginPage.navigate();
-        await loginPage.performLogin('standard_user', 'secret_sauce');
-        await inventoryPage.addItemToCartById(0);
-        expect(await inventoryPage.getNumberOfItemsInCart()).toBe('1');
+  const filterArr = ["az", "za", "lohi", "hilo"];
+ 
 
-        await inventoryPage.shopingCart.click();
-        expect(await shopingCartPage.cartItems.count()).toBeGreaterThanOrEqual(1);
 
-        await shopingCartPage.removeCartItemById(0);
-        await expect(shopingCartPage.cartItems).not.toBeAttached();
+  for (let element of filterArr) {
+    test(`Perform and verify sorting on the Inventory page by ${element}`, async ({
+      inventoryPage,
+    }) => {
+      //sorting names
+      const elementsNames =  await inventoryPage.productElementName.allTextContents();
+      const expectedAZ = await elementsNames.sort();
+     
+
+      //sorting prices
+      const elementsPrice = await inventoryPage.productPrice.allTextContents();
+      const elementsNumberOfPrice = await inventoryPage.selectPriceNumber(elementsPrice);
+      const expectedLoHi = await elementsNumberOfPrice.sort(function (a, b) { return a - b; });
+   
+
+      await inventoryPage.sortContainer.selectOption(element);
+      const catualElementsName = await inventoryPage.productElementName.allTextContents();
+      const actualElementsPrice = await inventoryPage.productPrice.allTextContents();
+      const actualElementsNumberOfPrice = await inventoryPage.selectPriceNumber(actualElementsPrice);
+
+      if (element === "az") {
+        expect(catualElementsName ).toEqual(expectedAZ);
+      } else if (element === "za") {
+        expect(catualElementsName).toEqual(expectedAZ.reverse());
+      } else if (element === "lohi") {
+         expect(actualElementsNumberOfPrice).toEqual(expectedLoHi );
+      } else if (element === "hilo") { 
+        expect(actualElementsNumberOfPrice).toEqual( expectedLoHi.reverse() );
+      }
```

Comment:
Lets update this test. Here we have sorting names block, but we dont need it in half of cases, same with sorting prices. For me it seems we'd better to have separate tests for sorting by name and by price. In this case we would have less if clauses in test and it will look more readable. Also Array.sort() doesn't need await, its sync function